astropy/io/fits/tests/test_image.py

```diff
@@ -73,7 +73,7 @@ def test_constructor_copies_header(self):
         # Original header should be unchanged
         assert phdr['FILENAME'] == 'labq01i3q_rawtag.fits'
 
-    @raises(ValueError)
+    @raises(IndexError)
```

Comment:
@embray - Thanks for reviewing this! You're right about the exception message that should be more helpful, I will implement this. But I'm also confused by your last sentence, is it possible to have `_read_all = True` if the file is not completely read ?